config/initializers/new_framework_defaults_7_1.rb

```diff
@@ -1,223 +0,0 @@
-
-# Enable validating only parent-related columns for presence when the parent is mandatory.
-# The previous behavior was to validate the presence of the parent record, which performed an extra query
-# to get the parent every time the child record was updated, even when parent has not changed.
-# Rails.application.config.active_record.belongs_to_required_validates_foreign_key = false
-
-# Enable precompilation of `config.filter_parameters`. Precompilation can
-# improve filtering performance, depending on the quantity and types of filters.
-# Rails.application.config.precompile_filter_parameters = true
-
-# Enable before_committed! callbacks on all enrolled records in a transaction.
-# The previous behavior was to only run the callbacks on the first copy of a record
-# if there were multiple copies of the same record enrolled in the transaction.
-# Rails.application.config.active_record.before_committed_on_all_records = true
-
-# Disable automatic column serialization into YAML.
-# To keep the historic behavior, you can set it to `YAML`, however it is
-# recommended to explicitly define the serialization method for each column
-# rather than to rely on a global default.
-# Rails.application.config.active_record.default_column_serializer = nil
-
-# Enable a performance optimization that serializes Active Record models
-# in a faster and more compact way.
-#
-# To perform a rolling deploy of a Rails 7.1 upgrade, wherein servers that have
-# not yet been upgraded must be able to read caches from upgraded servers,
-# leave this optimization off on the first deploy, then enable it on a
-# subsequent deploy.
-# Rails.application.config.active_record.marshalling_format_version = 7.1
-
-# Run `after_commit` and `after_*_commit` callbacks in the order they are defined in a model.
-# This matches the behaviour of all other callbacks.
-# In previous versions of Rails, they ran in the inverse order.
-# Rails.application.config.active_record.run_after_transaction_callbacks_in_order_defined = true
-
-# Whether a `transaction` block is committed or rolled back when exited via `return`, `break` or `throw`.
-#
-# Rails.application.config.active_record.commit_transaction_on_non_local_return = true
-
-# Controls when to generate a value for <tt>has_secure_token</tt> declarations.
-#
-# Rails.application.config.active_record.generate_secure_token_on = :initialize
-
-# ** Please read carefully, this must be configured in config/application.rb **
-# Change the format of the cache entry.
-# Changing this default means that all new cache entries added to the cache
-# will have a different format that is not supported by Rails 7.0
-# applications.
-# Only change this value after your application is fully deployed to Rails 7.1
-# and you have no plans to rollback.
-# When you're ready to change format, add this to `config/application.rb` (NOT
-# this file):
-#   config.active_support.cache_format_version = 7.1
```

Comment:
the comment is a little confusing, but I wonder if it means that it should be added to `config/application.rb` during the transition (ie: while you are enabling some defaults, before doing the change for `config.load_defaults 7.2`)

because i don't see this config in a new rails app, and the rails code seems to set it automatically when you change to 7.1 defaults https://github.com/rails/rails/blob/main/railties/lib/rails/application/configuration.rb#L309

Reply:
> because i don't see this config in a new rails app, and the rails code seems to set it automatically when you change to 7.1 defaults https://github.com/rails/rails/blob/main/railties/lib/rails/application/configuration.rb#L309

Yeah same for me

> the comment is a little confusing, but I wonder if it means that it should be added to config/application.rb during the transition (ie: while you are enabling some defaults, before doing the change for config.load_defaults 7.2)

That actually makes sense!